src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -35,9 +35,10 @@ public static String findIndexWithDimensionAndType(Pinecone pinecone, int dimens
         }
 
         for (IndexModel indexModel : indexModels) {
-            if (indexModel.getDimension() == dimension
-                && (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD) && indexModel.getSpec().getPod() != null)
-                || (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_SERVERLESS) && indexModel.getSpec().getServerless() != null)
+            // Sparse-dense is only supported with DOTPRODUCT
```

Comment:
Even though we create new indexes of type DOT PRODUCT in this class when an index doesn't exist, the chances are there can be an index in console that was manually created or via other means but has a metric COSINE or EUCLIDEAN. This can be an issue for sparse-dense vectors tests since they are only supported with DOT PRODUCT metric

Reply:
Is `findIndexWithDimensionAndType` only used for sparse-dense index tests? I'm not really understanding why we needed to add `indexModel.getMetric() == IndexMetric.DOTPRODUCT` to this.

Reply:
Is Cosine or Euclidean metric stricly needed for any other tests?